src/dune_engine/build_system.mli

```diff
@@ -12,6 +12,32 @@ type caching =
   ; check_probability : float
   }
 
+module Error : sig
+  type t
+
+  val message : t -> User_message.t
+end
+
+module Handler : sig
```

Comment:
Documentation please. One should be able to understand what `Start` or `Fail` mean without having to look at the implementation.